widget/widgetAssets/js/recordDistribution.service.js

```diff
@@ -79,8 +79,13 @@
             }
             angular.forEach(itemRecords, function (itemRecord) {
               var recordJson = {};
-              recordJson.name = $interpolate('{{ ' + config.titleField + ' }}')(itemRecord);
-              if (recordJson.name && recordJson.name === '') {
+              if (config.titleField) {
```

Comment:
Can you please see following logic will work?
recordJson.name = $interpolate(entity.displayTemplate)(itemRecord);
              if (config.titleField && $interpolate('{{ ' + config.titleField + ' }}')(itemRecord) != '') {
                recordJson.name = $interpolate('{{ ' + config.titleField + ' }}')(itemRecord);
              }

Reply:
It is added so that previous version didn't get impacted